app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return view("insert");
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $post= new Post;
+        $post->title= $request->get('title');
+        $post->body= $request->get('body');
+        $post->save();
+        return redirect("show");
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\Post  $post
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Post $post)
+    {
+        $posts = Post::all();
+        return view('show', ['posts'=>$posts]);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  \App\Models\Post  $post
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Post $post, $id)
+    {
+        $posts = Post::find($id);
+        return view('edit',['posts'=>$posts]);
+
+
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \App\Models\Post  $post
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, Post $post, $id)
+    {
+        $posts = Post::find($id);
+        $posts->title= $request->get('title');
+        $posts->body= $request->get('body');
+        $posts->save();
+        return redirect("show");
+
+    }
```

Comment:
Tính đến tình huống ko tìm thấy post nữa thì sao?
Dùng hàm Update cho gọn nè và tương tự trên thêm dòng trắng dãn cách nha
```suggestion
        $posts = Post::findOrFail($id);
        $posts->update($request->all());
        $posts->save();

        return redirect("show");
    }
```